thread-manager/src/lib.rs

```diff
@@ -0,0 +1,380 @@
+use {
+    anyhow::Ok,
+    serde::{Deserialize, Serialize},
+    std::{collections::HashMap, ops::Deref, sync::Arc},
+};
+
+pub mod native_thread_runtime;
+pub mod policy;
+pub mod rayon_runtime;
+pub mod tokio_runtime;
+
+pub use {
+    native_thread_runtime::{JoinHandle, NativeConfig, NativeThreadRuntime},
+    policy::CoreAllocation,
+    rayon_runtime::{RayonConfig, RayonRuntime},
+    tokio_runtime::{TokioConfig, TokioRuntime},
+};
+pub type ConstString = Box<str>;
+
+#[derive(Default, Debug)]
+pub struct ThreadManagerInner {
```

Comment:
does this need to be pub?

Reply:
Yes, else it can not be used in deref, which really hurts code that uses the structure. Unless there is a better way which I am not aware of.